src/store/tport/actions.js

```diff
@@ -80,6 +81,83 @@ export const updateTeleports = async function ({ commit }, account) {
+            let contractTrxs = res.data.transactions.filter((trx) => trx.to === contractHash);
+            var tokenTransfersInfo = [];
+            for (var i = 0; i < contractTrxs.length; i++) {
+                var trx = contractTrxs[i];
+                var logs = trx.logs;
+                for (var j = 0; j < logs.length; j++) {
+                    var log = logs[j];
+                    var data = log.data;
+                    var topics = log.topics;
+                    var iface = new ethers.utils.Interface(contractAbi);
+                    try {
+                        var tokenTrxReturn = iface.parseLog({ data, topics });
+                        function hex_to_ascii(str1) {
+                            var hex  = str1.toString();
+                            var str = '';
+                            for (var n = 0; n < hex.length; n += 2) {
+                                str += String.fromCharCode(parseInt(hex.substr(n, 2), 16));
+                            }
+                            return str;
+                        }
+                        if (tokenTrxReturn.name == "Teleport") {
+                            tokenTransfersInfo.push({...tokenTrxReturn.args, ref: trx.hash});
+                        }
+                    }
+                    catch (error) {
+
+                    }
+                }
+            }
+            var refs = tokenTransfersInfo.map((tokenTransferInfo) => { return tokenTransferInfo.ref});
+            let receipts = await this.$api.getTableRows({
+                code: process.env.TPORT_ADDRESS,
+                scope: process.env.TPORT_ADDRESS,
+                table: "receipts",
+                key_type: "i64",
+                limit: 10000,
+                reverse: true,
+                show_payer: false,
+            });
+            var obj = {};
+            receipts.rows.map((row)=>{
+                if (refs.includes(`0x${row.ref}`)) {
+                    var newRowObj = {...row};
+                    delete newRowObj.ref;
+                    obj[row.ref] = newRowObj;
+                }
+            });
+            console.log(obj);
+            let tokenTransfersInfoFinal = tokenTransfersInfo.map((tti) => {
+                var ref = tti.ref.replace("0x","");
+                return {
+                    ...tti,
+                    quantity: obj[ref].quantity,
+                    date: obj[ref].date,
+                    // chain_id: obj[ref].from_chain_id,
+                };
+            });
+            commit("setEvmTransactions",{transactions:tokenTransfersInfoFinal});
+            commit("setEvmTransactionsUpdating",false);
```

Comment:
Well done! I haven't checked your algorithm in detail, but it looks quite nice! :slightly_smiling_face: